src/webapi/html_elements/canvas.rs

```diff
@@ -0,0 +1,108 @@
+impl IElement for CanvasElement {}
+impl IHtmlElement for CanvasElement {}
+
+reference_boilerplate! {
+    CanvasElement,
+    instanceof HTMLCanvasElement
+    convertible to EventTarget
+    convertible to Node
+    convertible to Element
+    convertible to HtmlElement
+}
+
+/// Trait implemented by rendering contexts which can be obtained from a canvas.
+pub trait RenderingContext: FromReference {
+    /// Name which identifies this kind of rendering context.
+    const CONTEXT_TYPE: &'static str;
+}
+
+impl CanvasElement {
+    /// Returns a positive integer reflecting the height HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 150 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn height( &self ) -> u32 {
+        js! (
+            return @{self}.height;
+        ).try_into().unwrap()
+    }
+
+    /// Sets a positive integer reflecting the height HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 150 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn set_height( &self, value: u32 ) {
+        js! { @(no_return)
+            @{self}.height = @{value};
+        }
+    }
+
+    /// Returns a positive integer reflecting the width HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 300 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn width( &self ) -> u32 {
+        js! (
+            return @{self}.width;
+        ).try_into().unwrap()
+    }
+
+    /// Sets a positive integer reflecting the width HTML attribute of the <canvas> element
+    /// interpreted in CSS pixels. When the attribute is not specified, or if it is set to an
+    /// invalid value, like a negative, the default value of 300 is used.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/height)
+    pub fn set_width( &self, value: u32 ) {
+        js! { @(no_return)
+            @{self}.width = @{value};
+        }
+    }
+
+    /// Returns a drawing context on the canvas, or None if the context identifier is not supported.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext)
+    pub fn get_context<T: RenderingContext>( &self ) -> Option<T> {
+        js! (
+            return @{self}.getContext(@{T::CONTEXT_TYPE});
+        ).try_into().ok().and_then(|r| T::from_reference(r))
+    }
```

Comment:
I think this might not be flexible enough. E.g. for WebGL at some point in time in various browsers you could use one of the following names for what effectively was the same kind of a context: `"webgl", "experimental-webgl", "webkit-3d", "moz-webgl"`, so this mechanism should support specifying a list. (Or perhaps we should just have the `RenderingContext` export a `get_context` method and just delegate calling `getContext` to it completely?)

Reply:
OK, I will give `RenderingContext` a `from_canvas` method or something...